mtags/src/main/scala/scala/meta/internal/pc/OverrideCompletions.scala

```diff
@@ -0,0 +1,461 @@
+   * @return the list of TextEdit of both method implementations and auto imports.
+   */
+  private def implementAll(
+      typed: Tree,
+      range: l.Range,
+      t: Template,
+      text: String,
+      shouldAddOverrideKwd: Boolean,
+      isCandidate: Symbol => Boolean
+  ): List[l.TextEdit] = {
+    val overrideMembers = getMembers(
+      typed,
+      range,
+      t.pos,
+      text,
+      true,
+      false,
+      isCandidate
+    )
+
+    val allAbstractMembers = overrideMembers
+      .filter(_.sym.isAbstract)
+
+    val (allAbstractEdits, allAbstractImports) = toEdits(allAbstractMembers)
+
+    if (allAbstractEdits.length > 0) {
+
+      // infer necessary indent
+      //
+      // |object Test {
+      // |    class Foo extends Bar {} // inferred to 4
+      // |}
+      val lineStart = t.pos.source.lineToOffset(t.pos.line - 1)
+      val necessaryIndent = inferIndent(lineStart, text)
+
+      // infer indent for implementations
+      // if there's declaration in the class/object, follow its indent.
+      // otherwise the indent default to 2
+      val indent = typed.tpe.decls
+        .filter(sym =>
+          !sym.isSynthetic &&
+            !sym.isPrimaryConstructor &&
+            sym.pos.line != t.pos.line // filter out explicit primary constructor `class Foo(x: Int) ...`
+        )
+        .headOption
+        .map(existing => {
+          " " * inferIndent(
+            t.pos.source.lineToOffset(existing.pos.line - 1),
+            text
+          )
+        })
+        .getOrElse {
+          " " * (necessaryIndent + DefaultIndent)
+        }
+
+      // if the both opening/closing braces located in a line:
+      //
+      // object {
+      //   class Foo extends Bar {}
+      // }
+      //
+      // add an newline and indent in the end of implementations, so that
+      // the closing brace is indented.
+      //
+      // object {
+      //   class Foo extends Bar {
+      //     override def foo = ???
+      //   }
+      // }
+      val end =
+        if (t.pos.source.offsetToLine(t.pos.start) ==
+            t.pos.source.offsetToLine(t.pos.end)) "\n" + " " * necessaryIndent
+        else ""
+
+      val implementAll = new l.TextEdit(
+        range,
+        allAbstractEdits
+          .map(_.getNewText)
+          .mkString(
+            s"\n${indent}",
+            s"\n${indent}",
+            s"$end"
+          )
+      )
+      implementAll :: allAbstractImports.toList
+    } else {
+      Nil
+    }
+  }
+
+  /**
+   * Get the position of the opening brace of given Template.
+   * insert implementations onto the top of the body.
+   *
+   * @param text the text of the original source code.
+   * @param t the enclosing template for the class/object/trait we are implementing.
+   */
+  private def inferEditPosition(text: String, t: Template): Position = {
+    val start = t.pos.start
+    val end = t.pos.end
+    val offset = text.indexOf('{', start) + 1
+    t.pos.withStart(offset).withEnd(offset)
```

Comment:
Find the position of the opening brace for the Template. Since `Template` always has both the opening and closing braces, it's safe, I guess.

Does anybody have a better idea to find a good position to insert implements? 🤔